### PR TITLE
[ADD]business_requirement_versioning module

### PR DESCRIPTION
**Pull request by lonelysun**
replace #270

### CODE REVIEW

File: business_requirement_versioning/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    'name': 'Business Requirement Versioning',
+    'category': 'Business Requirements Management',
+    'summary': 'update the busuness requirement version',
+    'version': '10.0.1.0.0',
+    'website': 'https://www.elico-corp.com/',
+    "author": "Elico Corp, Odoo Community Association (OCA)",
+    'depends': [
+        'business_requirement',
+    ],
+    'data': [
```
Comment (elicoidal) [inline business_requirement_versioning/__manifest__.py:14]: Missing demo data (one example is enough)

**Comment by elicoidal** [comment]
@gurneyalex I rebuilt the runbot but can get it up: https://runbot.odoo-community.org/runbot/build/3313645
Could you help?

**Comment by elicoidal** [comment]
@lonelysun I have a traceback in Runbot (http://3320993-270-68c6d0.runbot1.odoo-community.org/web):
1. Go to BR0006, Set as Approved
2. Click on Renew
3. Get the following:
Traceback (most recent call last):
  File "/.repo_requirements/odoo/odoo/http.py", line 641, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/.repo_requirements/odoo/odoo/http.py", line 683, in dispatch
    result = self._call_function(**self.params)
  File "/.repo_requirements/odoo/odoo/http.py", line 333, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/service/model.py", line 101, in wrapper
    return f(dbname, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/http.py", line 326, in checked_call
    result = self.endpoint(*a, **kw)
  File "/.repo_requirements/odoo/odoo/http.py", line 941, in __call__
    return self.method(*args, **kw)
  File "/.repo_requirements/odoo/odoo/http.py", line 506, in response_wrap
    response = f(*args, **kw)
  File "/home/odoo/odoo-10.0/addons/web/controllers/main.py", line 895, in call_button
    action = self._call_kw(model, method, args, {})
  File "/home/odoo/odoo-10.0/addons/web/controllers/main.py", line 883, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/.repo_requirements/odoo/odoo/api.py", line 689, in call_kw
    return call_kw_multi(method, model, args, kwargs)
  File "/.repo_requirements/odoo/odoo/api.py", line 680, in call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/home/odoo/build/OCA/business-requirement/business_requirement_versioning/models/business_requirement_versioning.py", line 81, in renew_br
    version=count
  File "/home/odoo/build/OCA/business-requirement/business_requirement_versioning/models/business_requirement_versioning.py", line 53, in copy
    return super(BusinessRequirementVersion, self).copy(vals)
  File "/home/odoo/odoo-10.0/addons/pad/models/pad.py", line 132, in copy
    return super(PadCommon, self).copy(default)
  File "/.repo_requirements/odoo/odoo/models.py", line 4403, in copy
    vals = self.copy_data(default)[0]
  File "/home/odoo/odoo-10.0/addons/mail/models/mail_thread.py", line 299, in copy_data
    return super(MailThread, self.with_context(mail_notrack=True)).copy_data(default=default)
  File "/.repo_requirements/odoo/odoo/models.py", line 4324, in copy_data
    lines = [rec.copy_data()[0] for rec in self[name].sorted(key='id')]
TypeError: 'NoneType' object has no attribute '__getitem__'

File: business_requirement_versioning/models/business_requirement_versioning.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    def _compute_related_br_count(self):
+        for record in self:
+            count = len(record.env['business.requirement'].search(
+                [('source_id', '=', record.id)]))
+            record.related_br_count = count
+
+    @api.model
+    def _get_states(self):
+        res = super(BusinessRequirementVersion, self)._get_states()
+        res.append(('renewed', 'Renewed'))
```
Comment (elicoidal) [inline business_requirement_versioning/models/business_requirement_versioning.py:34]: renewed -> updated

File: business_requirement_versioning/tests/__init__.py [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# -*- coding:utf-8 -*-
+
+from . import test_renew
```
Comment (elicoidal) [inline business_requirement_versioning/tests/__init__.py:3]: renew->update

File: business_requirement_versioning/views/button_view.xml [hunk 4]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <record id="view_business_requirement_form" model="ir.ui.view">
+        <field name="name">business.requirement.form</field>
+        <field name="model">business.requirement</field>
+        <field name="inherit_id" ref="business_requirement.view_business_requirement_form"/>
+        <field name="type">form</field>
+        <field name="arch" type="xml">
+            <xpath expr="//header/field[@name='state']" position="before">
+                <button string="Renew" name="renew_br"
```
Comment (elicoidal) [inline business_requirement_versioning/views/button_view.xml:10]: Renew -> update (everywhere)

**Review by elicoidal — CHANGES_REQUESTED** [review]

**Comment by elicoidal** [comment]
@lonelysun please have a look

File: business_requirement_versioning/__manifest__.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# © 2017 Elico Corp (https://www.elico-corp.com).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+{
+    'name': 'Business Requirement Versioning',
+    'category': 'Business Requirements Management',
+    'summary': 'update the busuness requirement version',
+    'version': '10.0.1.0.0',
+    'website': 'https://www.elico-corp.com/',
```
Comment (nikul-serpentcs) [inline business_requirement_versioning/__manifest__.py:9]: @lonelysun IMHO change URL as per [this](https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L10)

File: business_requirement_versioning/views/button_view.xml [hunk 6]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <record id="view_business_requirement_form" model="ir.ui.view">
+        <field name="name">business.requirement.form</field>
+        <field name="model">business.requirement</field>
+        <field name="inherit_id" ref="business_requirement.view_business_requirement_form"/>
+        <field name="type">form</field>
+        <field name="arch" type="xml">
+            <xpath expr="//header/field[@name='state']" position="before">
```
Comment (nikul-serpentcs) [inline business_requirement_versioning/views/button_view.xml:9]: @lonelysun IMO Directly use `<field name="state" position="before">`

File: business_requirement_versioning/models/business_requirement_versioning.py [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+    version = fields.Integer(string='Version', default=0)
+    related_br_count = fields.Integer(
+        string='Related BR', copy=False, store=False,
+        compute='_compute_related_br_count'
+    )
+
+    @api.multi
+    def _compute_related_br_count(self):
+        for record in self:
+            count = len(record.env['business.requirement'].search(
```
Comment (nikul-serpentcs) [inline business_requirement_versioning/models/business_requirement_versioning.py:27]: @lonelysun IMO here use `search_count` method,  So Directly use count.

**Comment by elicoidal** [comment]
@lonelysun Let me know if you can finish this

**Comment by elicoidal** [comment]
@ruter-lyu can you finish this one too and port it to v12?

**Comment by ruter-lyu** [comment]
> @ruter-lyu can you finish this one too and port it to v12?

When I finished other migrations I'll start to do this

**Comment by pedrobaeza** [comment]
Closing by inactivity

**Pull request closed without merging**